feat(event-gateway): Architecture diagram + hybrid mode info (#5149)
Kong/developer.konghq.com: app/event-gateway/architecture.md

  The control plane generates data plane certificates and pushes configuration updates to the proxy nodes.
  It never sees the actual Kafka message payloads.
* **Data plane (self-managed)**: The data plane (DP) consists of stateless proxy nodes running within your own cluster.
  These nodes intercept Kafka client traffic, evaluate it against the policies pushed by the control plane, and proxy the allowed traffic to the backend Kafka brokers.

Periodically, the data plane polls the control plane for configuration updates.

Depending on the type of configuration update, the connection between the Kafka client and the backend can be affected:
* Updates to virtual cluster policies don't cause a connection drop. Policies reload dynamically and take effect on the next request.
* Updates to any other part of the configuration (for example, listener policies, auth, or namespaces in virtual clusters) cause a connection drop.
When the data plane receives configuration updates, it restarts the running proxy services.

If a connection drop occurs, the Kafka client is designed to handle short-lived breaks in connections.

> The components of the high-level architecture can be visualized like this:

Reviewer: ```suggestion
When the data plane receives configuration updates, it restarts the proxy services.

The Kafka client is designed to handle short-lived connection drops.

The following diagram illustrates the high-level architecture:
```

   subgraph EGW ["{{site.event_gateway_short}} data plane"]
      Analytics["Virtual cluster: <br/>analytics<br/>policies: e.g. ACL, filter"]
      Payments["Virtual cluster: <br/>payments<br/>policies: e.g. ACL,<br> Schema, Filter"]
   end

   BackendKafka["Backend Kafka<br/>cluster"]
   KafkaClient<-->EGW<-->BackendKafka

   OB["Observability system<br/>metrics & logs"]
   EGW--OTEL<br/>exporter-->OB
end

style Konnect stroke-dasharray:3
style Customer stroke-dasharray:3

{% endmermaid %}
<!--vale on-->

_**Figure 1**: The control plane (CP) is fully managed in {{site.konnect_short_name}}. When the data plane (DP) polls the CP for configuration, the CP pushes the config to the self-managed DP.
The DP proxies Kafka client traffic through virtual clusters to backend Kafka clusters, and exports metrics and logs to an observability system via OpenTelemetry._

### {{site.event_gateway_short}} entities

In {{site.event_gateway_short}}, an entity is a component or object that makes up the {{site.event_gateway_short}} and its ecosystem. 
Entities represent the various building blocks used to configure and manage {{site.event_gateway_short}}, and each entity has a specific role.

> All the configurations of the entities that run on the data plane live in the control plane. 

Reviewer: ```suggestion
Configuration for entities running on the data plane is stored in the control plane.
```
